### PR TITLE
test: Improve API snapshots

### PR DESCRIPTION
**Pull request by mattsjones**

### CODE REVIEW

File: generate-component-docs/generate.ts [hunk 1]
```diff
@@ -9,6 +9,14 @@ const propBlacklist = ['key'];
 const tsconfigPath = path.join(__dirname, '../tsconfig.json');
 const componentsFile = path.join(__dirname, '../lib/components/index.ts');
 
+const reactNodeType =
+  'string | number | boolean | {} | ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<any, any, any>)> | ReactNodeArray | ReactPortal';
```
Comment (mattsjones) [inline generate-component-docs/generate.ts]: This is a bit of a hack and won't work if the type changes in any way. I think just keep an eye on it as it's much cleaner with the shorthand.

File: generate-component-docs/generate.ts [hunk 2]
```diff
@@ -9,6 +9,14 @@ const propBlacklist = ['key'];
...
+
+const stringAliases: Record<string, string> = {
+  [reactNodeType]: 'ReactNode',
+  boolean: 'boolean',
```
Comment (markdalgleish) [inline generate-component-docs/generate.ts]: Why is this needed?

Comment (mattsjones) [reply generate-component-docs/generate.ts]: This was basically already in the code. Technically it seems a Boolean is a union of true and false. This just makes it show as the word boolean.
